pennylane/transforms/resolve_dynamic_wires.py

```diff
@@ -0,0 +1,222 @@
+
+    Args:
+        tape (QuantumScript): A circuit that may contain dynamic wire allocations and deallocations
+        zeroed (Sequence[Hashable]): a register of wires known to be the zero state
+        any_state (Sequence[Hashable]): a register of wires with any state
+        min_int (Optional[int]): If not ``None``, new wire labels can be created starting at this
+            integer and incrementing whenever a new wire is needed.
+
+    Returns:
+        tuple[QuantumScript], Callable[[ResultBatch], Result]: A batch of tapes and a postprocessing function
+
+    .. note::
+
+        This transform currently uses a "Last In, First Out" (LIFO) stack based approach to distributing wires.
+        This minimizes the total number of wires used, at the cost of higher depth and more resets. Other
+        approaches could be taken as well, such as a "First In, First out" algorithm that minimizes depth.
+
+        This approach also means we pop wires from the *end* of the stack first.
+
+    For a dynamic wire requested to be in the zero state (``require_zeros=True``), we try three things before erroring:
+
+      #. If wires exist in the ``zeroed`` register, we take one from that register
+      #. If no ``zeroed`` wires exist, we pull one from ``any_state`` and apply a reset operation
+      #. If no wires exist in the ``zeroed`` or ``any_state`` registers, we increment ``min_int`` and
+         add a new wire
+
+    For a dynamic wire with ``require_zeros=False``, we try:
+
+      #. If wires exist in the ``any_state``, we take one from that register
+      #. If no wires exist in ``any_state``, we pull one from ``zeroed``
+      #. If no wires exist in the ``zeroed`` or ``any_state`` registers, we increment ``min_int`` and
+         add a new wire
+
+    This approach minimizes the width of the circuit at the cost of more reset operations.
+
+    .. code-block:: python
+
+        def circuit(require_zeros=True):
+            with qml.allocation.allocate(1, require_zeros=require_zeros) as wires:
+                qml.X(wires)
+            with qml.allocation.allocate(1, require_zeros=require_zeros) as wires:
+                qml.Y(wires)
+
+    >>> print(qml.draw(circuit)())
+    <DynamicWire>: ──Allocate──X──Deallocate─┤
+    <DynamicWire>: ──Allocate──Y──Deallocate─┤
+
+    If we provide two zeroed qubits to the transform, we can see that the two operations have been
+    assigned to both wires known to be in the zero state.
+
+    >>> assigned_two_zeroed = qml.transforms.resolve_dynamic_wires(circuit, zeroed=("a", "b"))
+    >>> print(qml.draw(assigned_two_zeroed)())
+    a: ──Y─┤
+    b: ──X─┤
+
+    If we only provide one zeroed wire, we perform a reset on that wire before reusing for the ``Y`` operation.
+
+    >>> assigned_one_zeroed = qml.transforms.resolve_dynamic_wires(circuit, zeroed=("a", "b"))
+    >>> print(qml.draw(assigned_one_zeroed)())
+    b: ──X──┤↗│  │0⟩──Y─┤
```

Comment:
```suggestion
    >>> assigned_one_zeroed = qml.transforms.resolve_dynamic_wires(circuit, zeroed=("a",))
    >>> print(qml.draw(assigned_one_zeroed)())
    a: ──X──┤↗│  │0⟩──Y─┤
```

Reply:
Isn't this supposed to be the code example where you pass only one `zeroed` wire, as stated in the sentence before and in the name of the transformed function? 🤔